tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala

```diff
@@ -89,10 +89,17 @@ object VarDictJava extends Configuration {
   /** Plain text tumor-normal separator. */
   val TumorNormalSeparator: String = "|"
 
-  /** Pulls a sample name out of a BAM file. */
-  private[vc] def extractSampleName(bam: PathToBam): String = {
+  /** Return the first read group in a BAM. */
+  private[vc] def firstReadGroup(bam: PathToBam): Option[SAMReadGroupRecord] = {
+    import com.fulcrumgenomics.commons.CommonsDef.javaIteratorAsScalaIterator
     val in = SamReaderFactory.make().open(bam)
-    yieldAndThen(in.getFileHeader.getReadGroups.iterator().next().getSample) { in.close() }
+    yieldAndThen(in.getFileHeader.getReadGroups.iterator.toStream.headOption) { in.close() }
+  }
+
+  /** Return the sample name from the first read group in a BAM. */
+  private[vc] def extractSampleName(bam: PathToBam): Option[String] = {
+    val rg = firstReadGroup(bam).getOrElse(throw new IllegalStateException(s"No read groups found in BAM: $bam"))
+    Option(rg.getSample)
```

Comment:
On second thought, since you require on `327` to have the sample name ALSO defined, why not just make this method return `String` and put that check here?
```suggestion
    rg.getSample match {
      case null => throw new IllegalStateException(s"The first read group's sample name does not exist: $bam"))
      case sn  => sn
    }
```
or use `Option(rg.getSemple).getOrElse(...)`

Reply:
I initially favored deferred evaluation for both functions. I tried the refactor and like it, I think it keeps the the code that calls the function simpler.